tests/verify-role-results.yml

```diff
@@ -11,8 +8,36 @@
     var: _storage_volumes_list
   when: _storage_volumes_list | length > 0
 
-- name: Ensure the specified devices in the role were correctly configured in the user's system
-  include_tasks: "test-{{ test }}.yml"
-  loop: "{{ storage_tests_list }}"
+#
+# Verify pools and the volumes they contain.
+#
+- name: Verify the volumes listed in storage_pools were correctly managed
+  include_tasks: "test-verify-pool.yml"
+  loop: "{{ _storage_pools_list }}"
   loop_control:
-    loop_var: test
+    loop_var: current_pool
+  when: _storage_pools_list is defined and _storage_pools_list | length > 0
+
+- name: Clean up variable namespace
+  set_fact:
+    current_pool: null
+
+#
+# Verify standalone volumes.
+#
+- name: Verify the volumes with no pool were correctly managed
+  include_tasks: "test-verify-volume.yml"
+  loop: "{{ _storage_volumes_list }}"
+  loop_control:
+    loop_var: current_volume
+  when: _storage_volumes_list is defined and _storage_volumes_list | length > 0
+
+#
+# Clean up.
+#
+- name: Clean up variable namespace
+  set_fact:
+    storage_test_fstab: null
+    storage_test_blkinfo: null
+    current_pool: null
+    current_volume: null
```

Comment:
good idea, since there is no way to unset a variable or fact and using the task file multiple times from the same playbook would otherwise yield unexpected results. Would it work to register `storage_test_blkinfo` always, instead of only if `storage_test_blkinfo is not defined or not storage_test_blkinfo`? This could be an alternative to setting it to `null`.

Reply:
The intent is to collect the information once for all devices per test/validation run. Since I'm collecting the info in a tasks file that applies to a single volume (`test-verify-volume.yml`) I want to do so conditionally to prevent re-running it unnecessarily. I could collect the info at the top level, but it would be less clear in `verify-role-results.yml` what the information is being collected for. So I think this could be addressed either with a comment in `test-verify-volume.yml` or by moving the invocation of `blockdev_info` up into `verify-role-results.yml`.

I also plan to abstract the test-verification-specific facts (probably into a single dict) so that the top-level can reset them all without having to know what facts have been defined/used.